### PR TITLE
Separated upgrade jirik

### PR DESCRIPTION
**Pull request by index-git**
- ~~Tests~~
- ~~Layman Test Client~~
- [x] Changlelog
- [x] Documentation

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -30,6 +30,20 @@ stop-demo:
 build-demo:
 	docker-compose -f docker-compose.deps.demo.yml -f docker-compose.demo.yml build layman layman_client geoserver timgen
 
+upgrade-demo:
+	mkdir -p layman_data deps/qgis/data
+	docker-compose -f docker-compose.deps.demo.yml -f docker-compose.demo.yml build layman layman_client geoserver timgen
```
Comment (index-git) [inline Makefile:35]: `layman_client` is redundant here

File: Makefile [hunk 2]
```diff
@@ -30,6 +30,20 @@ stop-demo:
...
+	docker-compose -f docker-compose.deps.demo.yml -f docker-compose.demo.yml run --rm --no-deps -u root layman bash -c "cd src && python3 -B setup_geoserver.py"
+	docker-compose -f docker-compose.deps.demo.yml -f docker-compose.demo.yml up -d --force-recreate --no-deps postgresql qgis geoserver redis timgen layman_client nginx
```
Comment (index-git) [inline Makefile:37]: `layman_client` is redundant here

File: Makefile [hunk 3]
```diff
@@ -30,6 +30,20 @@ stop-demo:
...
+	docker-compose -f docker-compose.deps.demo.yml -f docker-compose.demo.yml run --rm --no-deps layman bash -c "cd src && python3 layman_flush_redis.py && python3 wait_for_deps.py && python3 standalone_upgrade.py"
+
+upgrade-demo-full:
+	mkdir -p layman_data deps/qgis/data
+	docker-compose -f docker-compose.deps.demo.yml -f docker-compose.demo.yml build layman layman_client geoserver timgen
```
Comment (index-git) [inline Makefile:42]: `layman_client` is redundant here

File: Makefile [hunk 4]
```diff
@@ -30,6 +30,20 @@ stop-demo:
...
+	docker-compose -f docker-compose.deps.demo.yml -f docker-compose.demo.yml run --rm --no-deps -u root layman bash -c "cd src && python3 -B setup_geoserver.py"
+	docker-compose -f docker-compose.deps.demo.yml -f docker-compose.demo.yml up -d --force-recreate --no-deps postgresql qgis geoserver redis timgen layman_client micka nginx
```
Comment (index-git) [inline Makefile:44]: `layman_client` is redundant here

File: Makefile [hunk 5]
```diff
@@ -74,6 +88,12 @@ restart-celery-dev:
 	docker-compose -f docker-compose.deps.yml -f docker-compose.dev.yml up --no-deps -d flower
 	docker-compose -f docker-compose.deps.yml -f docker-compose.dev.yml up --no-deps -d layman_client
 
+upgrade-dev:
+	mkdir -p layman_data layman_data_test tmp deps/qgis/data
+	docker-compose -f docker-compose.deps.yml -f docker-compose.dev.yml run --rm --no-deps -u root layman_dev bash -c "cd src && python3 -B setup_geoserver.py"
```
Comment (index-git) [inline Makefile:93]: `layman_client` is redundant here

File: Makefile [hunk 6]
```diff
@@ -74,6 +88,12 @@ restart-celery-dev:
...
+	docker-compose -f docker-compose.deps.yml -f docker-compose.dev.yml up -d timgen layman_client postgresql qgis nginx-qgis geoserver redis micka
```
Comment (index-git) [inline Makefile:94]: `layman_client` is redundant here

File: Makefile (hunk 1 above)
Comment (jirik) [reply Makefile:35]: We need it because of wait_for_deps

File: Makefile (hunk 2 above)
Comment (jirik) [reply Makefile:37]: We need it because of wait_for_deps

File: Makefile (hunk 4 above)
Comment (jirik) [reply Makefile:44]: We need it because of wait_for_deps

File: Makefile (hunk 3 above)
Comment (jirik) [reply Makefile:42]: We need it because of wait_for_deps

File: Makefile (hunk 5 above)
Comment (jirik) [reply Makefile:93]: We need it because of wait_for_deps

File: Makefile (hunk 6 above)
Comment (jirik) [reply Makefile:94]: We need it because of wait_for_deps

File: README.md [hunk 7]
```diff
@@ -127,6 +127,10 @@ General steps to upgrade layman to MINOR or PATCH version:
    - `layman_data`
    - `deps/*/data` (data directories of external dependencies)
 1. Follow **Upgrade requirements** in [Changelog](CHANGELOG.md) of all MINOR and PATCH versions greater than your current version and lower or equal to the version you are upgrading to.
+1. If you are expecting long-running upgrade, run **standalone upgrade**. The command depends on how you are starting Layman.
+   - If you are starting Layman with `make start-demo`, run `make upgrade-demo`.
+   - If you are starting Layman with `make start-demo-full` or `make start-demo-full-with-optional-deps`, run `make upgrade-demo-full`.
+   - If you are starting Layman with `make start-dev`, you don't need to run standalone migration.
```
Comment (index-git) [inline README.md:133]: Add reasoning for a separate upgrade option. When to use it, what happened if admin forgot it.

**Review by jirik — APPROVED** [review]

**Pull request merged**